src/cpp/preprocessor/aie2/aie2_blob_preprocessor_input.h

```diff
@@ -65,11 +65,12 @@ class aie2_blob_preprocessor_input : public preprocessor_input
   std::vector<uint32_t> pm_id_list;
   bool haspreempt = false;
   virtual uint32_t extractSymbolFromBuffer(std::vector<char>& mc_code, const std::string& section_name, const std::string& argname) = 0;
-  void aiecompiler_json_parser(const boost::property_tree::ptree& pt);
+  void aiecompiler_json_parser(const boost::property_tree::ptree& pt, const std::string& id);
   void dmacompiler_json_parser(const boost::property_tree::ptree& pt);
-  void readmetajson(std::istream& patch_json);
-  void extract_control_packet_patch(const std::string& name, const uint32_t arg_index, const boost::property_tree::ptree& _pt);
-  void extract_coalesed_buffers(const std::string& name, const boost::property_tree::ptree& _pt);
+  void readmetajson(std::istream& patch_json, const std::string& id);
+  void extract_control_packet_patch(const std::string& name, const uint32_t arg_index,
```

Comment:
const is not useful here.
```suggestion
  void extract_control_packet_patch(const std::string& name, uint32_t arg_index,
```